include/sound/sof/dai-intel.h

```diff
@@ -82,6 +82,7 @@ struct sof_ipc_dai_ssp_params {
 struct sof_ipc_dai_hda_params {
 	struct sof_ipc_hdr hdr;
 	uint32_t link_dma_ch;
+	uint32_t reserved[8];
```

Comment:
It should be added in sof_ipc_pcm_params for you assign link dma channel in hw_param.
The sof_ipc_dai_hda_params  is for dai_link loading

Reply:
@RanderWang I suppose i didnt do that good of a job explaining things. Because of our requirement to couple before setting the HDA format, the host DMA stream tag and the link DMA stream tag will always need to be the same. So, I only send one stream tag to the DSP during host pcm hw_params. Because of the way, the link DMA tags are "reserved", the link dma channel assigned during link hw_params will always be the same as host dma stream tag. 

Please give this PR a try and test all kinds of scenarios you think could be broken. That's really what I am doing too. @fredoh9 and myself have been testing a few kinds of concurrent use cases and havent been able to see any issues with our hda generic topology so far.

Reply:
yes, I just said in FW patch.  two cases:
(1) for two FEs share one BE, the param would be called two times and one of them the tag should be different.
(2) For some streams with only BE like tone generator, there is no host stream

Reply:
@RanderWang when the hw_params will be the called the second time, we will ignore the channel as it will not be INVALID at that point.

Reply:
yes, I get it.